tools/format.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+if [[ ! -e tools/format.sh ]]; then
+  echo "This tool must be run from the topmost directory." >&2
+  exit 1
+fi
+
+set -e
+
+FIND="find . -name .git -prune -o -name _deps -prune -o -name .build -prune -o"
+
+# Correct common miscapitalizations.
+sed -i 's/Open[t]elemetry/OpenTelemetry/g' $($FIND -type f -print)
+# No CRLF line endings.
+sed -i 's/\r$//' $($FIND -type f -print)
+# No trailing spaces.
+sed -i 's/ \+$//' $($FIND -type f -print)
+
+CMD=clang-format-8
```

Comment:
Do we need to require clang-format-8? Could we use just clang-format here?

Reply:
That's what I started with, but on Debian stable clang-format is v5.0.0, which is too old to understand `AfterExternBlock` in our `.clang-format`

Reply:
Can we rely on the CI to catch an old clang-format? Or do a version check and error out if it's too low?

I have clang-format 8, but os x doesn't have a clang-format-8 alias.

Reply:
Resolved by making the script more complicated. :^)